File: src/main/java/net/moltendorf/Bukkit/IntelliDoors/IntelliDoors.java
package net.moltendorf.Bukkit.IntelliDoors;

import net.moltendorf.Bukkit.IntelliDoors.listener.Interact;
import net.moltendorf.Bukkit.IntelliDoors.listener.Redstone;
import org.bukkit.plugin.PluginManager;
import org.bukkit.plugin.java.JavaPlugin;

/**
 * Created by moltendorf on 15/05/23.
 *
 * @author moltendorf
 */
public class IntelliDoors extends JavaPlugin {
  // Main instance.
  private static IntelliDoors instance = null;

  protected static IntelliDoors getInstance() {
    return instance;
  }

  // Variable data.
  protected Settings settings = null;

  @Override
  public void onEnable() {
    instance = this;

    // Construct new settings.
    settings = new Settings();

    // Register listeners.
    PluginManager manager = getServer().getPluginManager();

    manager.registerEvents(new Interact(), this);
    manager.registerEvents(new Redstone(), this);
  }

  @Override
  public void onDisable() {
    instance = null;
  }

  public static void DebugByte(byte bite) {
    getInstance().getServer().broadcastMessage(String.format("%4s", Integer.toBinaryString(bite)).replace(' ', '0'));
  }

  public static void DebugString(String string) {
    getInstance().getServer().broadcastMessage(string);
  }
}

File: src/main/kotlin/net/moltendorf/Bukkit/IntelliDoors/IntelliDoors.kt
package net.moltendorf.Bukkit.IntelliDoors

import net.moltendorf.Bukkit.IntelliDoors.listener.Interact
import net.moltendorf.Bukkit.IntelliDoors.listener.Redstone
import org.bukkit.plugin.java.JavaPlugin

/**
 * Created by moltendorf on 15/05/23.

 * @author moltendorf
 */
class IntelliDoors : JavaPlugin() {
  // Variable data.
  lateinit var settings: Settings
    private set

  override fun onEnable() {
    instance = this

    // Construct new settings.
    settings = Settings()

    // Are we enabled?
    enabled = settings.enabled

    if (enabled) {
      // Register listeners.
      val manager = server.pluginManager

      if (settings.interact) {
        manager.registerEvents(Interact(), this)
      }

      if (settings.redstone) {
        manager.registerEvents(Redstone(), this)
      }
    }
  }

  override fun onDisable() {
    enabled = false
  }

  companion object {
    var enabled = false
      private set

    lateinit var instance: IntelliDoors
      private set
  }
}
